appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/WebPrincipal.java

```diff
@@ -173,7 +176,7 @@ private static String getPrincipalName(X509Certificate[] certificates, SecurityC
         // Use the full DN name from the certificates. This should normally be the same as 
         // context.getCallerPrincipal(), but a realm could have decided to map the name in which
         // case they will be different.
-        return certificates[0].getSubjectX500Principal().getName();
+        return certificates[0].getSubjectX500Principal().getName(X500Principal.RFC2253, CertificateRealm.oidMap);
```

Comment:
We are using `getName(X500Principal.RFC2253, CertificateRealm.oidMap)` several times. Make a utility method so that X500Principal representation is only in 1 place for future?

Reply:
I'm not sure that's really worth it? Unless I'm misunderstanding you we'd just be replacing `certificates[0].getSubjectX500Principal().getName(X500Principal.RFC2253, CertificateRealm.oidMap)` with `CertificateRealm.getX500PrincipalSubjectName(certificates[0])`.

It seems fragile to me to store the name of the certificate somewhere outside of the actual certificate itself.

Also, it isn't strictly necessary for me to put the full X500Principal.RFC2253 stuff everywhere, it's me just being safe. From my testing as long as you get the name in this manner from the `CertificateRealm` it propagates outward such that simply doing `getName()` will still have the OIDs translated.